addons/website_event_meet/controllers/main.py

```diff
@@ -0,0 +1,163 @@
+        )
+
+    def _event_meeting_rooms_get_values(self, event, lang=None, open_room_id=None):
+        # meeting_rooms = event.sudo().meeting_room_ids.filtered(lambda m: m.active)
+        search_domain = self._get_event_rooms_base_domain(event)
+        meeting_rooms_all = request.env['event.meeting.room'].sudo().search(search_domain)
+        if lang:
+            search_domain = expression.AND([
+                search_domain,
+                [('room_lang_id', '=', int(lang))]
+            ])
+        meeting_rooms = request.env['event.meeting.room'].sudo().search(search_domain)
+        meeting_rooms = meeting_rooms.sorted(lambda m: (m.is_pinned, m.room_last_joined, m.id), reverse=True)
+
+        return {
+            # event information
+            "event": event.sudo(),
+            'main_object': event,
+            # rooms
+            "meeting_rooms": meeting_rooms,
+            "current_lang": request.env["res.lang"].browse(int(lang)) if lang else False,
+            "available_languages": meeting_rooms_all.mapped("room_lang_id"),
+            "default_lang_code": request.env.user.lang,
+            "open_room_id": int(open_room_id) if open_room_id else None,
+            # environment
+            "is_event_manager": request.env.user.has_group("event.group_event_manager"),
+        }
+
+    @http.route(["/event/create_meeting_room"], type="http", auth="public", methods=["POST"], website=True)
+    def create_meeting_room(self, **post):
+        name = post.get("name")
+        summary = post.get("summary")
+        target_audience = post.get("audience")
+        lang_code = post.get("lang_code")
+        max_capacity = post.get("capacity")
+        event_id = int(post.get("event"))
+
+        # get the record to be sure they really exist
+        event = request.env["event.event"].browse(event_id).exists()
+        lang = request.env["res.lang"].search([("code", "=", lang_code)], limit=1)
+
+        if not event or not event.can_access_from_current_website():
+            raise Forbidden()
+
+        if not event.website_published or not lang or max_capacity == "no_limit":
+            raise Forbidden()
+
+        _logger.info("New meeting room (%s) create by %s" % (name, request.httprequest.remote_addr))
+
+        meeting_room = request.env["event.meeting.room"].sudo().create(
+            {
+                "name": name,
+                "summary": summary,
+                "target_audience": target_audience,
+                "is_pinned": False,
+                "event_id": event.id,
+                "room_lang_id": lang.id,
+                "room_max_capacity": max_capacity,
+            },
+        )
+
+        return redirect("/event/%s/meeting_rooms?open_room_id=%i" % (slug(event), meeting_room.id))
+
+    @http.route(["/event/active_langs"], type="json", auth="public")
+    def active_langs(self):
+        return request.env["res.lang"].sudo().get_installed()
```

Comment:
we could use "/web/dataset/call_kw/res.lang/get_installed" route instead of creating a new one